File: API/Entities/Hotel.cs

using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace API.Entities
{
    public class Hotel
    {
        [Key][DatabaseGenerated(System.ComponentModel.DataAnnotations.Schema.DatabaseGeneratedOption.Identity)]
        public int Id {get; set;}
        public string Nume {get; set;}
        public string Adresa {get; set;}
        public decimal PricePerNight{get; set;}
        public decimal Rating{get;set;}
        public int CityId {get; set;}
        public City City {get; set; }
        public List<Photo> Photos { get; set; } = new();
        public ICollection<Review> Reviews {get; set;}
    }
}

File: client/src/app/_models/hotel.ts
import { Photo } from "./photo";
import { Review } from "./review";

export interface Hotel{
    nume: string;
    pricePerNight?: number;
    pricePerNightCameraTripla?: number;
    rating?: number;
    adresa:string;
    cityId: number;
    photoUrl?: string;
    photos?: Photo[];
    reviews?: Review[];
    id?:number;
}
export class HotelFilter{
    pageNumber: number = 1;
    pageSize: number = 5;
    sortOrder: number = 1;
    sortField: string = "nume";
    nume?: string;
    pricePerNight?: number;
    pricePerNightCameraTripla?: number;
    rating?: number;
    adresa?:string;
    cityId?: number;
    photoUrl?: string;
    photos?: Photo[];
    reviews?: Review[];
    id?:number;
    nrPersoane = 0;
    buget = 0;
    startDate = '';
    endDate = '';
    nrNopti = 0;
}